scripts/merge_duplicate_users.py

```diff
@@ -0,0 +1,38 @@
+from django.db.models.functions import Lower
+from django.db.models import Count
+
+from bluebottle.members.models import Member
+
+from bluebottle.clients.models import Client
+from bluebottle.clients.utils import LocalTenant
+
+from bluebottle.activities.models import Activity, Contributor
+from bluebottle.initiatives.models import Initiative
+
+
+for client in Client.objects.all():
+    with LocalTenant(client):
+        duplicate = Member.objects.annotate(
+            lower=Lower('email')
+        ).values('lower').annotate(count=Count('lower')).filter(count__gt=1)
+        for result in duplicate:
+            first = Member.objects.filter(email__iexact=result['lower']).order_by('date_joined').first()
+            for duplicate in Member.objects.filter(
+                email__iexact=result['lower']
+            ).exclude(pk=first.pk).order_by('date_joined'):
+                for activity in Activity.objects.filter(owner=duplicate):
+                    activity.owner = first
+                    activity.execute_triggers(send_messages=False)
+                    activity.save()
+
+                for contributor in Contributor.objects.filter(user=duplicate):
+                    contributor.user = first
+                    contributor.execute_triggers(send_messages=False)
+                    contributor.save()
+
+                for initiative in Initiative.objects.filter(owner=duplicate):
+                    initiative.owner = first
+                    initiative.execute_triggers(send_messages=False)
+                    initiative.save()
```

Comment:
Are we skipping all other user related objects? Wall posts, user activity, permissions, themes, skills, avatar etc? Not sure how big this issue is though.

Reply:
Is it a wild guess that the first sign-up is the account that is used the most or did you investigate this? Which one has the most activity? Again, not sure how much effort we should put into this. Maybe keep some reference to the one user account we picked in the anonymised one? Abuse the about_me field to print 'duplicate of #id'? Then we can always repair it afterwards..
.

Reply:
or `duplicate.email = '{}-duplo-{}'.format(first.email, duplicate.id)`
Needs another save though...

Reply:
> Are we skipping all other user related objects? Wall posts, user activity, permissions, themes, skills, avatar etc? Not sure how big this issue is though.

I was not exactly sure what other things to migrate. Wallpost seems logical. Permissions not so much since admin accounts should not be part of this, and neither should partner accounts. 

Themes, skills and avatars I do not really know. I do not think there will be many of those.

Reply:
> or `duplicate.email = '{}-duplo-{}'.format(first.email, duplicate.id)`
> Needs another save though...

Jep. I added a way to identify merged users now.

Reply:
> Is it a wild guess that the first sign-up is the account that is used the most or did you investigate this? Which one has the most activity? Again, not sure how much effort we should put into this. Maybe keep some reference to the one user account we picked in the anonymised one? Abuse the about_me field to print 'duplicate of #id'? Then we can always repair it afterwards..
> .

Yes. I did not find a single account that had more activity on the newer account. Most accounts had no activity on the second. In the end it does not really matter though. As long as the older stuff is merged.